src/remake.jl

```diff
@@ -50,6 +50,14 @@ function remake(prob::ODEProblem; f = missing,
                 _kwargs...)
     if u0 === missing
         u0 = prob.u0
+    else
+        if u0 isa Dict || eltype(u0) isa Pair
+            if hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :states)
+                u0 = handle_varmap(u0, prob.f.sys, field = :states, defaults = prob.f.sys.defaults)
```

Comment:
Shouldn't the default be constructed from the current states and parameters that the problem holds?